pkg/sanity/tests.go

```diff
@@ -40,17 +44,20 @@ func DescribeSanity(text string, body func(*SanityContext)) bool {
 // registerTestsInGinkgo invokes the actual Gingko Describe
 // for the tests registered earlier with DescribeSanity.
 func registerTestsInGinkgo(sc *SanityContext) {
+	roleMatch := regexp.MustCompile(fmt.Sprintf(".*(?:%s).*", strings.Join(sc.Config.Roles.String(), "|")))
 	for _, test := range tests {
-		Describe(test.text, func() {
-			BeforeEach(func() {
-				sc.setup()
-			})
+		if roleMatch.MatchString(test.text) {
```

Comment:
Would it make sense to use
```
if !roleMatch.MatchString(test.text) {
    continue
}
````
to avoid indenting the rest of the code?

Just thinking aloud here, I am not sure whether it is considered better Go style.

Reply:
No idea about "better" Go style. I avoided it for my own personal aesthetics, since I like to avoid explicit flow control statements where possible. :)

Reply:
I've been asked a few times to rewrite code such that it indents less, so having explicit "return" or "continue" statements might indeed be the preferred style.

Anyway, it makes the diff smaller, so please change it as suggested above unless it hurts your aesthetics too much ;-}

Reply:
idiomatic go code (as well as my preferred style) is to have explicit returns/continues to avoid having large blocks of code indented. It makes it very clear under "what condition" the function should terminate early and not continue doing any execution whatsoever in that function. With large blocks of indented code you run the risk of having the scroll down before noticing that there may be more code after the closing brace of the indented code which hurts readability.

Reply:
But such code tends to lead to higher cyclomatic complexity, which then leads to a tendency to have microfunctions as a way to combat that, which further hurts readability.

Reply:
I don't believe this changes the cyclomatic complexity. It merely inverts the cases, does not add or take away any

Reply:
I never claimed it did.